eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/immutable/AbstractImmutableSortedBag.java

```diff
@@ -461,4 +462,12 @@ public void reverseForEachWithIndex(ObjectIntProcedure<? super T> procedure)
     {
         throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".asReversed() not implemented yet");
     }
+
+    @Override
+    public ImmutableSortedSet<T> selectUnique()
```

Comment:
I would suggest to move this implementation to `ImmutableSortedBagImpl` and have implementation using `forEachWithOccurrences` similar to other comments.
The `SortedSet` should inherit the comparator from ImmutableSortedBagImpl.

For ImmutableEmptySortedBag, simply return an empty ImmutableSortedSet which also inherits the comparator.